### PR TITLE
Upgrade celery to 5.4.0

### PR DESCRIPTION
**Pull request by gzurowski**
- Upgrade celery dependencies to 5.4.0.
- Replace and relocate deprecated configuration for scheduled task from decorators to the celery configuration.

### CODE REVIEW

**Review by koutst — APPROVED** [review]
LGTM. One nit is that it is sort of nice to have the definition of when the task is running right above the task.

**Comment by gzurowski** [comment]
> LGTM. One nit is that it is sort of nice to have the definition of when the task is running right above the task.

This is a result of the changes around scheduled tasks in celery 5.x. Task schedules are no longer tied to the functions with decorators but is added to the celery configuration (https://docs.celeryq.dev/en/stable/userguide/periodic-tasks.html#crontab-schedules).

**Review by yattias — APPROVED** [review]

File: src/researchhub/celery.py [hunk 1]
```diff
@@ -38,93 +37,126 @@
...
+    "reputation_recalc-hot-score-for-open-bounties": {
+        "task": "reputation.tasks.recalc_hot_score_for_open_bounties",
+        "schedule": crontab(hour=12, minute=0),
+        "options": {
+            "priority": 4,
+            "queue": QUEUE_BOUNTIES,
+        },
+    },
+    # User
+    "user_execute-editor-daily-payout-task": {
```
[the remark this answers is not in the document]
Comment (gzurowski) [reply src/researchhub/celery.py:146]: It's scheduled to launch at 23:05

[the remark this answers is not in the document]
Comment (gzurowski) [reply src/researchhub/celery.py:146]: @yattias This is fixed in #1784.

**Pull request merged**